macOS ARM64 Build Failure (constexpr Lanes)

## Error 
```cpp
loops.cpp:417:20: error: constexpr variable 'L' must be initialized by a constant expression
  constexpr size_t L = Lanes(d);
```

## Root Cause
https://github.com/exaloop/codon/blob/c234f873cc0fee77119685de38f9d307589f863e/codon/runtime/numpy/loops.cpp#L5-L7

- The current guard disables SVE only on Linux, but on macOS ARM64 (Apple Silicon) this guard is not applied, so the SVE target remains enabled. (Apple Silicon does not supported 'SVE')
- For SVE, Highway's `Lanes()` is not `constexpr` because vector length is determined at runtime.

## Possible Fix 
Disable SVE on ARM64 platforms that do not support SVE.

```cpp
// Before
#if defined(__linux__) && (defined(__aarch64__) || defined(__arm64__))

// After
#if (defined(__aarch64__) || defined(__arm64__)) && \
    !defined(__ARM_FEATURE_SVE)
#define HWY_DISABLED_TARGETS HWY_ALL_SVE
#endif
```

--- a/cmake/deps.cmake
+++ b/cmake/deps.cmake
@@ -172,7 +172,7 @@ endif()
 CPMAddPackage(
     NAME highway
     GITHUB_REPOSITORY "google/highway"
-    GIT_TAG 1.2.0
+    GIT_TAG 1.3.0
     EXCLUDE_FROM_ALL YES
     OPTIONS "HWY_ENABLE_CONTRIB ON"
             "HWY_ENABLE_EXAMPLES OFF"

--- a/codon/runtime/numpy/loops.cpp
+++ b/codon/runtime/numpy/loops.cpp
@@ -152,7 +152,7 @@ struct CosFunctor {
   static inline auto vector(const hn::ScalableTag<T> d, const V &v) {
     // Values outside of [-LIMIT, LIMIT] are not valid for SIMD version.
     const T LIMIT = limit<T>();
-    constexpr size_t L = hn::Lanes(d);
+    HWY_LANES_CONSTEXPR size_t L = hn::Lanes(d);
     T tmp[L];
     Store(v, d, tmp);
 
@@ -330,7 +330,7 @@ struct SinFunctor {
   static inline auto vector(const hn::ScalableTag<T> d, const V &v) {
     // Values outside of [-LIMIT, LIMIT] are not valid for SIMD version.
     const T LIMIT = limit<T>();
-    constexpr size_t L = hn::Lanes(d);
+    HWY_LANES_CONSTEXPR size_t L = hn::Lanes(d);
     T tmp[L];
     Store(v, d, tmp);
 
@@ -367,7 +367,7 @@ struct SinhFunctor {
   static inline auto vector(const hn::ScalableTag<T> d, const V &v) {
     // Values outside of [-LIMIT, LIMIT] are not valid for SIMD version.
     const T LIMIT = limit<T>();
-    constexpr size_t L = hn::Lanes(d);
+    HWY_LANES_CONSTEXPR size_t L = hn::Lanes(d);
     T tmp[L];
     Store(v, d, tmp);
 
@@ -414,7 +414,7 @@ struct HypotFunctor {
 template <typename T, typename F>
 void UnaryLoop(const T *in, size_t is, T *out, size_t os, size_t n) {
   const hn::ScalableTag<T> d;
-  constexpr size_t L = Lanes(d);
+  HWY_LANES_CONSTEXPR size_t L = hn::Lanes(d);
   T tmp[L];
   size_t i;
 
@@ -449,7 +449,7 @@ template <typename T, typename F>
 void BinaryLoop(const T *in1, size_t is1, const T *in2, size_t is2, T *out, size_t os,
                 size_t n) {
   const hn::ScalableTag<T> d;
-  constexpr size_t L = Lanes(d);
+  HWY_LANES_CONSTEXPR size_t L = hn::Lanes(d);
   T tmp1[L];
   T tmp2[L];
   size_t i;

